### ISSUE TITLE
Support for polymorph objects in Model classes

### ISSUE THREAD

**Issue by tom-ioki** [ioki-mobility/ioki-ruby#113]
## Problem

As I am working on the `task_list` endpoints for the operator API, I noticed that the object for the attributes `start_location` and `end_location` can either be a `Place` or a `Station`.  The problem with that is that the model classes only support one `class_name`. This means when the defined `class_name` for `start_location` is `Place` but the serialized value is of type `Station` the gem is not able to create the `TaskList` model correctly.

## Idea

Add support for arrays for the `class_name` setting.
```ruby
attribute :start_location,
          on:         :read,
          type:       :object,
          class_name: ['Place', 'Station']
```

To achieve this the `Ioki::Model::Base` class needs a little add on. Something like that:
```ruby
        if class_name.is_a?(Array) && !value.dig('type').nil?
          serialized_object = value['type'].split('_').map(&:capitalize).join('')
          if class_name.include?(serialized_object)
            class_name = serialized_object
          end
        end
```

What this does is, it checks if the type of an attribute's serialized object is contained in the defined class_name array. If that is the case the type value will be titelized and used as class_name. The rest of the Base class stays the same.

**Comment by eikes** [comment]
Good point! We can make use of the camelize method here:

```ruby
        if class_name.is_a?(Array) && !value.dig('type').nil?
          value_class_name = Support.camelize value['type']
          if class_name.include?(object_class_name)
            class_name = value_class_name
          end
        end
```

Alternatively we could, by default, use the `value['type']` to infer the class and fall back to the `class_name` if it can't be inferred

**Comment by tom-ioki** [comment]
closed with #112
